### DISCUSSION TITLE
Make image viewable in camera?

### DISCUSSION THREAD

**Discussion by feklee** [ImageMagick/ImageMagick#2349] [General]
The plan is to convert maps to JPEG images that I can view inside of my Ricoh GR III digicam. This is useful when I don’t have my phone with me.

For now, I create a test image in ImageMagick and try to make it viewable in the phone. So far I have not succeeded, although the Exif data looks very similar. The image [R0000349.JPG][3] was shot with the camera. Here is what I tried:

    $ convert -size 6000x4000 radial-gradient:red-blue -sampling-factor 4:2:2 S0000422.JPG
    $ exiftool -TagsFromFile R0000349.JPG -all:all -unsafe -XML:All S0000422.JPG
    $ exiftool R0000349.JPG | sort >R0000349.EXIF
    $ exiftool S0000422.JPG | sort >S0000422.EXIF
    $ diff R0000349.EXIF S0000422.EXIF
    49,52c49,52
    < File Access Date/Time           : 2020:08:01 13:49:29+08:00
    < File Inode Change Date/Time     : 2020:08:01 13:49:29+08:00
    < File Modification Date/Time     : 2020:08:01 13:49:29+08:00
    < File Name                       : R0000349.JPG
    ---
    > File Access Date/Time           : 2020:08:01 13:49:35+08:00
    > File Inode Change Date/Time     : 2020:08:01 13:49:35+08:00
    > File Modification Date/Time     : 2020:08:01 13:49:35+08:00
    > File Name                       : S0000422.JPG
    54c54
    < File Size                       : 11 MB
    ---
    > File Size                       : 448 kB
    80a81
    > JFIF Version                    : 1.01
    145c146
    < Thumbnail Offset                : 32066
    ---
    > Thumbnail Offset                : 23362

The camera cannot display `S0000422.JPG`.

As a workaround, I found I can open and save `S0000422.JPG` using Sony’s [Imaging Edge Desktop][2]. Optionally, if I copy the Exif data using ExifTool as above, the image will still display in the camera. However, Imaging Edge Desktop does not run on Linux, and as far as I can tell, there is no batch processing option.

PS: I may crosspost this to the ExifTool forum. Interesting is the archived thread [Make JPEG file DCF compatible][1], but with no solution.

[1]: https://exiftool.org/forum/index.php?topic=2228.0
[2]: https://imagingedge.sony.net/en-us/ie-desktop.html
[3]: https://archive.org/details/r-0000349

**Comment by snibgo** [comment]
I suggest testing with R0000349.JPG. Upload that file to your camera, with a different name. Can the camera still read it? If you change something trivial in the file, can the camera still read it?

Look at the problem from another angle. Take a bunch of JPG files from the camera. What do they all have in common? Perhaps the ThumbnailOffset is constant, and this is what you need to make your file readable.

I suggest you ask the question on the https://discuss.pixls.us/ forum.

**Comment by feklee** [comment]
> I suggest testing with R0000349.JPG. Upload that file to your camera, with a different name. Can the camera still read it? If you change something trivial in the file, can the camera still read it?

There is a lot that can be changed, and the camera can still read the file. A name change is no problem. Note what I wrote about Imaging Edge Desktop. I can even make the generated file readable, but I want to be able to do that on the Linux command line.

> I suggest testing with R0000349.JPG. Upload that file to your camera, with a different name. Can the camera still read it? If you change something trivial in the file, can the camera still read it?

Thanks for the suggestion!

> I suggest you ask the question on the https://discuss.pixls.us/ forum.

Not sure I want to post in yet another place, unless there is more technical expertise than here. (which I doubt)

**Comment by snibgo** [comment]
I notice that "exiftool R0000349.JPG" has:
```
Compression                     : JPEG (old-style)
```
IM won't create files with old-style compression.

You have a process of elimination. You know your camera can read some files but not others. You can slowly step through all the variables to find which makes it unreadable.

**Comment by feklee** [comment]
Right, I found that too. *How is old-style JPEG different from just JPEG?*

**Comment by snibgo** [comment]
As far as I know, exiftool reporting "-Compression=JPEG (old-style)" in a JPG file just means that tag is present and has the value 6. Just for fun, we can change it to other values, such as 4, which is "T6/Group 4 Fax". This doesn't seem to prevent any software from correctly reading the file.

But there is a chance that your camera reads that tag and, if it isn't present or is not 6, the camera refuses to read the image.

To set the tag:
```
exiftool -args -n -Compression=6 r.jpg
```
Verify this:
```
exiftool -args r.jpg   | grep ompre

-Compression=JPEG (old-style)
```

See also https://en.wikipedia.org/wiki/TIFF#TIFF_Compression_Tag and the Exif standard at http://cipa.jp/std/documents/download_e.html?DC-008-Translation-2019-E

**Comment by feklee** [comment]
Thanks, but the tag is set, as can be seen in the `diff` found in my initial posting.

**Comment by feklee** [comment]
In the end, aside from changing chrome subsampling with ImageMagick, it is a job for ExifTool. First create a test image:

    convert -size 2000x1000 radial-gradient:red-blue TEST.JPG

Then process the test image, which includes copying Exif data from a picture [R0000349.JPG][1], which was taken with the camera:

    INPUT=TEST.JPG
    OUTPUT=A0000001.JPG # DCF compatible
    THUMBNAIL=THUMB.JPG
    ORIGINAL=R0000349.JPG # taken with the camera
    convert -sampling-factor 4:2:2 $INPUT $OUTPUT
    exiftool -TagsFromFile $ORIGINAL -all:all -unsafe -XML:All -JFIF:ALL= $OUTPUT
    convert $INPUT -resize 160x120 -background black -gravity center -extent 160x120 -sampling-factor 4:2:2 $THUMBNAIL
    exiftool "-ThumbnailImage<=$THUMBNAIL" $OUTPUT

The output needs to be placed in a [DCF][2] compatible path on the camera.

Make sure that the number in the file name is unique on the card, or else the camera may be confused. For example, if there is `A0000001.JPG` and `B0000001.JPG` on the card, then my Ricoh GR III displays neither of the images.

[1]: https://archive.org/details/r-0000349
[2]: https://en.wikipedia.org/wiki/Design_rule_for_Camera_File_system
